### PR TITLE
piglit: opengl test suite

### PR DESCRIPTION
**Pull request by dankegel**
Because inquiring minds want to know if OpenGL _really_ works in snaps.

### CODE REVIEW

**Comment by dankegel** [comment]
The failure appears to be spurious... is there a way to turn down the verbosity of the cmake
build, or allow a longer build log?

**Comment by dholbach** [comment]
Wow, this is an interesting build failure (https://s3.amazonaws.com/archive.travis-ci.org/jobs/166607459/log.txt):
```
The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over).

The job has been terminated
```
Confirming build locally.

File: piglit/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+should be replaced with a small wrapper script specific to
+this project).
+
+Example usage:
+
+First, do a very quick sanity check to make sure you can access the graphics card:
+
+$ piglit run tests/sanity results/sanity
+$ piglit summary console results/sanity
+
```
Comment (dholbach) [inline piglit/README.md]: Where are these tests shipped to? Can we note down the path here? Even if it's something like `/snap/piglit/current/<someplace>`?

File: piglit/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    #command: piglit
+    # but that doesn't find python, so:
+    #command: python3 piglit
+    # but that doesn't find the opengl drivers, so set LIBGL_DRIVERS_PATH:
+    #command: opengl-launch python3 $SNAP/bin/piglit
+    # but cmake sets binaries' rpath to /lib/piglit/lib, so add the missing $SNAP:
+    #command: env LD_LIBRARY_PATH=$SNAP/lib/piglit/lib:$LD_LIBRARY_PATH opengl-launch python3 $SNAP/bin/piglit
+    # but piglit throws UnicodeDecodeError, so:
+    command: env LC_ALL=C.UTF-8 LD_LIBRARY_PATH=$SNAP/lib/piglit/lib:$LD_LIBRARY_PATH opengl-launch python3 $SNAP/bin/piglit
+    plugs:
```
Comment (dholbach) [inline piglit/snapcraft.yaml]: Thanks for documenting this. This really should be easier. Do you know if there are bugs filed for some of the issues?

**Review by dholbach — COMMENTED** [review]
Nice work! Just two small questions.

File: piglit/README.md (hunk 1 above)
Comment (dankegel) [inline piglit/README.md]: I'll add a comment

File: piglit/snapcraft.yaml (hunk 2 above)
Comment (dankegel) [inline piglit/snapcraft.yaml]: I'll check and add comments as appropriate.

**Comment by dankegel** [comment]
Looking at dmesg while running shows I've missed a trick:

[82377.292349] audit: type=1400 audit(1476198649.841:408): apparmor="DENIED" operation="open" profile="snap.piglit.piglit" name="/dev/kmsg" pid=9898 comm="dmesg" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
[82377.311673] audit: type=1400 audit(1476198649.857:409): apparmor="DENIED" operation="open" profile="snap.piglit.piglit" name="/dev/kmsg" pid=9899 comm="dmesg" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
[82377.379139] audit: type=1400 audit(1476198649.925:410): apparmor="DENIED" operation="sendmsg" profile="snap.piglit.piglit" pid=9900 comm="shader_runner" family="unix" sock_type="dgram" protocol=0 requested_mask="send" denied_mask="send" addr=none peer_addr="@6E7669646961316138326630633100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" peer="unconfined"
[82377.704069] audit: type=1400 audit(1476198650.249:411): apparmor="DENIED" operation="open" profile="snap.piglit.piglit" name="/dev/kmsg" pid=9901 comm="dmesg" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0

I'll try to clean that up (once I figure out wtf :-) )

**Comment by dholbach** [comment]
Does http://askubuntu.com/questions/783979/how-do-i-debug-snaps help?

**Pull request closed without merging**